panoramisk/ami_protocol.py

```diff
@@ -27,11 +36,19 @@ def send(self, data, as_list=False):
             else:
                 klass = actions.Action
             data = klass(data, as_list=as_list)
-        self.transport.write(str(data).encode(encoding))
-        self.responses[data.id] = data
-        if data.action_id:
-            self.responses[data.action_id] = data
-        return data.future
+        if self.is_closing():
+            self.factory.awaiting_actions.put_nowait((data, as_list))
+        else:
+            try:
+                self.transport.write(str(data).encode(encoding))
+            except Exception as e:
+                self.factory.log.exception(e)
```

Comment:
you should have: self.factory.log.exception("Exception raised")
because, with this code, you'll see 2 times the same exception.